sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/IdentityClient.java

```diff
@@ -553,7 +555,12 @@ private Mono<AccessToken> getTokenFromMsalMIClient(String resource) {
                     throw new RuntimeException(e);
                 }
             }))
-            .onErrorMap(t -> new CredentialUnavailableException("Managed Identity authentication is not available.", t))
+            .onErrorMap(t -> {
+                if (options.isChained() && t instanceof MsalJsonParsingException) {
+                    return new CredentialUnavailableException("Managed Identity authentication is not available.", t);
+                }
+                return new ClientAuthenticationException("Managed Identity authentication is not available.", null, t);
```

Comment:
if MI is not available as per message, why not throw Unavilable Exception only ?

Reply:
Ah, I was going to change this message to something indicating getting a token failed. Will fix.